src/plist.c

```diff
@@ -1012,22 +1014,20 @@ Int             EqPlist (
         return 0L;
     }
 
+    CheckRecursionBefore();
+
     /* loop over the elements and compare them                             */
     for ( i = 1; i <= lenL; i++ ) {
         elmL = ELM_PLIST( left, i );
         elmR = ELM_PLIST( right, i );
-        if ( elmL == 0 && elmR != 0 ) {
-            return 0L;
-        }
-        else if ( elmR == 0 && elmL != 0 ) {
-            return 0L;
-        }
-        else if ( ! EQ( elmL, elmR ) ) {
+        if ( (elmL == 0 ) != (elmR == 0) || ! EQ( elmL, elmR ) ) {
```

Comment:
Can I be picky and ask for another pair of brackets here? I should know what the priority order of `!=` and `||` is, but I'm afraid I can't remember. However, I agree the rearranged code is fine.